apps/src/applab/AppLabTooltipOverlay.jsx

```diff
@@ -0,0 +1,148 @@
+/** @file App Lab-specific Tooltip Overlay */
+
+import TooltipOverlay, {coordinatesProvider} from '../templates/TooltipOverlay';
+import elementUtils from './designElements/elementUtils';
+import { scaledDropPoint } from './gridUtils';
+import { connect } from 'react-redux';
+import { ApplabInterfaceMode } from './constants';
+import { ellipsify } from '../utils';
+
+var ELEMENT_ID_TEXT_MAX_CHAR = 12;
+
+const AppLabTooltipOverlay = React.createClass({
+  propTypes: {
+    // width, height, mouseX and mouseY are given in app-space, not screen-space
+    width: React.PropTypes.number,
+    height: React.PropTypes.number,
+    mouseX: React.PropTypes.number,
+    mouseY: React.PropTypes.number,
+    // Provided by redux
+    isInDesignMode: React.PropTypes.bool.isRequired
+  },
+
+  getInitialState: () => ({
+    hoveredControlId: null
+  }),
+
+  componentDidMount() {
+    document.addEventListener('mousemove', this.onMouseMove);
+  },
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this.onMouseMove);
+  },
+
+  onMouseMove(event) {
+    this.setState({
+      hoveredControlId: this.getHoveredControlId(event.target)
+    });
+  },
+
+  /**
+   * Gets the element id of the Applab UI control user is hovering over, if any.
+   * If the user is in design mode, we strip the element id prefix.
+   * @param {EventTarget} eventTarget The mouseover event target
+   * @returns {string} id of the Applab UI control the mouse is over. Returns null if none exist.
+   * @private
+   */
+  getHoveredControlId(eventTarget) {
+    // Check that the element is a child of a screen
+    if (eventTarget && $(eventTarget).parents('div.screen').length > 0) {
+      var controlElement = eventTarget;
+
+      // Check to see the mouseover target is a resize handle.
+      // If so, grab the id of associated control instead of the resize handle itself.
+      // We need to do this because for very small controls, the resize handle completely
+      // covers the control itself, making it impossible to show the id tooltip
+      if (isResizeHandle(controlElement)) {
+        controlElement = getAssociatedControl(controlElement);
+      }
+
+      // If we're in design mode, get the element id without the prefix
+      if (this.props.isInDesignMode) {
+        return elementUtils.getId(controlElement);
+      }
+
+      return controlElement.id;
+    }
+
+    return null;
+  },
+
+  /**
+   * Internal helper to generate the element id string to display in tooltip.
+   * @returns {string}
+   * @private
+   */
+  getElementIdText() {
+    return "id: " + ellipsify(this.state.hoveredControlId, ELEMENT_ID_TEXT_MAX_CHAR);
```

Comment:
Ah... it doesn't [(example here)](http://xn--dahlstrm-t4a.net/svg/css/text-overflow-ellipsis.svg).  There are some suggestions for workarounds [here](http://stackoverflow.com/a/9249966/5000129) but this is probably okay for now.